config/config.go

```diff
@@ -39,10 +39,11 @@ const defaultSamplingProbability = 0.001
 
 // Configuration configures and creates Jaeger Tracer
 type Configuration struct {
-	Disabled   bool            `yaml:"disabled"`
-	Sampler    *SamplerConfig  `yaml:"sampler"`
-	Reporter   *ReporterConfig `yaml:"reporter"`
-	RPCMetrics bool            `yaml:"rpc_metrics"`
+	Disabled            bool            `yaml:"disabled"`
+	Sampler             *SamplerConfig  `yaml:"sampler"`
+	Reporter            *ReporterConfig `yaml:"reporter"`
+	RPCMetrics          bool            `yaml:"rpc_metrics"`
+	BaggageRestrictions bool            `yaml:"baggage_restrictions"`
```

Comment:
I made baggage restrictions opt in instead of opt out for now (keeps the size of this PR smaller because if I do opt out, I need to change all the tests). We probably should do opt out and I'll do it in a subsequent PR.